File: app/src/main/java/com/example/fruitsdiary/util/SchedulerProvider.java
package com.example.fruitsdiary.util;

import android.support.annotation.NonNull;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class SchedulerProvider {

    @NonNull
    public Scheduler computation(){
        return Schedulers.computation();
    }

    @NonNull
    public Scheduler io(){
        return Schedulers.io();
    }

    @NonNull
    public Scheduler ui(){
        return AndroidSchedulers.mainThread();
    }
}

File: app/src/main/java/com/example/fruitsdiary/util/SchedulerProvider.kt
package com.example.fruitsdiary.util

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

class SchedulerProvider {

    fun io(): Scheduler {
        return Schedulers.io()
    }

    fun ui(): Scheduler {
        return AndroidSchedulers.mainThread()
    }
}
